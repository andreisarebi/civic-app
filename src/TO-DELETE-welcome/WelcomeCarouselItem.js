import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, View } from 'react-native';
import colors from '../styles/colors';

const WelcomeCarouselItem = props => {
  return (
    <View style={stylesSmallScreen.container}>
      <View style={stylesSmallScreen.badge}>{props.image}</View>
      <Text style={stylesSmallScreen.title}>{props.title}</Text>
    </View>
  );
};

WelcomeCarouselItem.propTypes = {
  image: PropTypes.object,
  title: PropTypes.string,
};

// const stylesLargeScreen = StyleSheet.create({});

const stylesSmallScreen = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  badge: {
    height: 250,
    width: 250,
    marginTop: 24,
    borderRadius: 125,
    backgroundColor: colors.offWhite,
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.5,
    shadowRadius: 3,
  },
  title: {
    color: colors.white,
    fontSize: 20,
    textAlign: 'center',
    marginTop: 24,
  },
});

export default WelcomeCarouselItem;
