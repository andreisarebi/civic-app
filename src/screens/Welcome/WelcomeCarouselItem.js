import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, View } from 'react-native';
import colors from '../../styles/colors';

const WelcomeCarouselItem = props => (
  <View style={styles.container}>
    <View style={styles.badge}>{props.image}</View>
    <Text style={styles.title}>{props.title}</Text>
    {!props.isSmallScreen && <Text style={styles.subtitle}>{props.subtitle}</Text>}
  </View>
);

WelcomeCarouselItem.propTypes = {
  image: PropTypes.element.isRequired,
  isSmallScreen: PropTypes.bool.isRequired,
  subtitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  container: { alignItems: 'center' },
  badge: {
    height: 250,
    width: 250,
    marginTop: 24,
    borderRadius: 125,
    backgroundColor: colors.offWhite,
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.5,
    shadowRadius: 3,
  },
  subtitle: {
    color: colors.white,
    fontSize: 16,
    lineHeight: 24,
    paddingLeft: 20,
    paddingRight: 20,
    textAlign: 'center',
  },
  title: {
    color: colors.white,
    fontSize: 20,
    textAlign: 'center',
    marginTop: 24,
  },
});

export default WelcomeCarouselItem;
