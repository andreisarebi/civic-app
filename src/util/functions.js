import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
