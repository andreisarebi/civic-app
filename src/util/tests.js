import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
