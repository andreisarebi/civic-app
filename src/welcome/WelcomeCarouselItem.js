import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, View } from 'react-native';
import colors from '../styles/colors';

const WelcomeCarouselItem = props => {
  return (
    <View style={stylesSmallScreen.container}>
      <View {...props} style={stylesSmallScreen.badge}>
        {props.image}
      </View>
      <Text style={stylesSmallScreen.title}>{props.title}</Text>
      <Text style={stylesSmallScreen.subtitle}>{props.subtitle}</Text>
    </View>
  );
};

WelcomeCarouselItem.propTypes = {
  image: PropTypes.object,
  title: PropTypes.string,
  subtitle: PropTypes.string,
};

// const stylesLargeScreen = StyleSheet.create({});

const stylesSmallScreen = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  badge: {
    height: 250,
    width: 250,
    marginTop: 20,
    borderRadius: 125,
    backgroundColor: colors.offWhite,
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.5,
    shadowRadius: 3,
  },
  title: {
    color: colors.white,
    fontSize: 20,
    textAlign: 'center',
    paddingTop: 20,
    paddingBottom: 8,
  },
  subtitle: {
    color: colors.white,
    fontSize: 16,
    lineHeight: 24,
    paddingLeft: 20,
    paddingRight: 20,
    textAlign: 'center',
  },
});

export default WelcomeCarouselItem;
