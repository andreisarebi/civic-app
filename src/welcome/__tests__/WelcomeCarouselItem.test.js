import React from 'react';
import { Image } from 'react-native';
import { createSnapshot } from '../../util/tests';
import WelcomeCarouselItem from '../WelcomeCarouselItem';

describe('WelcomeCarouselItem', () => {
  it('renders correctly', () => {
    const props = {
      image: <Image />,
      title: 'Welcome Carousel',
      subtitle: 'Some more text',
    };

    const snapshot = createSnapshot(WelcomeCarouselItem, props);
    expect(snapshot).toMatchSnapshot();
  });
});
