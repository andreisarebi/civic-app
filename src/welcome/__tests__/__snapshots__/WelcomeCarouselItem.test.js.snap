// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeCarouselItem renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderRadius": 125,
        "height": 250,
        "justifyContent": "center",
        "marginTop": 24,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 3,
        "width": 250,
      }
    }
  >
    <Image />
  </Component>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 20,
        "marginTop": 24,
        "textAlign": "center",
      }
    }
  >
    Welcome Carousel
  </Text>
</Component>
`;
